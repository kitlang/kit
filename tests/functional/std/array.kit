function main() {
    printf("original\n");
    var a: Array[Int] = Array[Int].new(4);
    printf("length = %zu\n", a.length);
    a[0] = 11;
    a[1] = 23;
    a[2] = 35;
    a[3] = 47;
    // a = [1, 2, 3, 4];
    printf("first element = %i\n", a[0]);
    printf("second element = %i\n", a[1]);
    printf("third element = %i\n", a[2]);
    printf("fourth element = %i\n", a[3]);
    for x in a {
        printf("iterating: %i\n", x);
    }

    printf("copy\n");
    var a2: Array[Int] = a.copy();
    printf("length = %zu\n", a2.length);
    printf("first element = %i\n", a2[0]);
    printf("third element = %i\n", a2[2]);
    for x in a2 {
        printf("%i\n", x);
    }
}
