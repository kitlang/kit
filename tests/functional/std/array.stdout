original
length = 4
first element = 11
second element = 23
third element = 35
fourth element = 47
iterating: 11
iterating: 23
iterating: 35
iterating: 47
copy
length = 4
first element = 11
third element = 35
11
23
35
47
